File: 2024_04/0406/middle/change.c
#include<stdio.h>

int main(){
    int defaultMoney = 1000;
    int changesMoney = 0;
    int count = 0;
    int i = 0;
    int coin[6] = {500, 100, 50, 10, 5, 1};
    scanf("%d", &changesMoney);
    defaultMoney -= changesMoney;

    while(defaultMoney>0){
        if(defaultMoney >= coin[i]){
            defaultMoney-=coin[i]; count++;
        }else i++;
    }
    printf("%d", count);
    return 0;
}

File: 2024_04/0406/beginner/change.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int defaultMoney = 1000;
        int changesMoney = 0;
        int count = 0;
        int i = 0;
        int[] coin = {500, 100, 50, 10, 5, 1};

        changesMoney = scanner.nextInt();
        defaultMoney -= changesMoney;

        while (defaultMoney > 0) {
            if (defaultMoney >= coin[i]) {
                defaultMoney -= coin[i];
                count++;
            } else {
                i++;
            }
        }

        System.out.println(count);
        scanner.close();
    }
}
